### PR TITLE
Introduce a class that can't be used.

### PR DESCRIPTION
**Pull request by bangerth**
Specifically, declare DoFInvalidAccessor that is the accessor
for cases where dim>spacedim -- i.e., things that can't logically
happen, but that at times do happen in code like
```
  for (quad=0; quad<GeometryInfo<dim>::quads_per_cell; ++quad)
    cell->quad(quad)->do_something();
```
Here, the code is not executed in 1d, but we want it to be syntactically
correct so that we can compile the function. Having a DoFInvalidAccessor
then serves the same purpose that InvalidAccessor already does for
tria accessor functions: such objects cannot be created but rather
throw an exception, but they provide the correct syntax.

### CODE REVIEW

**Comment by bangerth** [comment]
ping

File: include/deal.II/dofs/dof_accessor.h [hunk 1]
```diff
@@ -1141,6 +1140,81 @@ class DoFAccessor<0,DoFHandlerType<1,spacedim>, level_dof_access> : public TriaA
 };
 
 
+
+/* -------------------------------------------------------------------------- */
+
+
+/**
+ * A class that represents DoF accessor objects to iterators that don't make sense
```
Comment (tamiko) [inline include/deal.II/dofs/dof_accessor.h]: to -> for ?

Comment (tamiko) [reply include/deal.II/dofs/dof_accessor.h]: Ah. Now I understand that sentence.

**Pull request merged**